astropy/units/format/vounit.py

```diff
@@ -124,24 +129,19 @@ def _get_unit(cls, t: LexToken) -> UnitBase:
             raise
 
     @classmethod
-    def _get_unit_name(cls, unit: NamedUnit) -> str:
+    def _decompose_to_known_units(
+        cls, unit: core.CompositeUnit | core.NamedUnit
+    ) -> UnitBase:
         # The da- and d- prefixes are discouraged.  This has the
         # effect of adding a scale to value in the result.
-        if isinstance(unit, core.PrefixUnit):
-            if unit._represents.scale == 10.0:
-                raise ValueError(
-                    f"In '{unit}': VOUnit can not represent units with the 'da' "
-                    "(deka) prefix"
-                )
-            elif unit._represents.scale == 0.1:
-                raise ValueError(
-                    f"In '{unit}': VOUnit can not represent units with the 'd' "
-                    "(deci) prefix"
-                )
-        name = unit._get_format_name(cls.name)
-        if name not in cls._custom_units:
-            cls._validate_unit(name, detailed_exception=True)
-        return name
+        if isinstance(unit, core.PrefixUnit) and unit._represents.scale in (0.1, 10.0):
+            return cls._decompose_to_known_units(unit._represents)
+        if (
+            isinstance(unit, core.NamedUnit)
+            and (name := unit._get_format_name(cls.name)) in cls._custom_units
```

Comment:
The assigned name is not used, so this can just be `unit._get_format_name(cls.name) in cls._custom_unit`